src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,44 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private Pair<K, V>[] pairs;
+
+    public StorageImpl() {
+        pairs = new Pair[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size(); i++) {
+            if (key == null && pairs[i].getKey() == null
+                    || pairs[i].getKey() != null && pairs[i].getKey().equals(key)) {
+                pairs[i].setValue(value);
+                return;
+            }
+        }
+        pairs[size()] = new Pair<>(key, value);
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size(); i++) {
+            if (key == null && pairs[i].getKey() == null
+                    || pairs[i].getKey() != null && pairs[i].getKey().equals(key)) {
+                return pairs[i].getValue();
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        int counter = 0;
+        for (int i = 0; i < MAX_SIZE; i++) {
+            if (pairs[i] != null) {
+                counter++;
+            }
+        }
+        return counter;
```

Comment:
```suggestion
        return counter;
```
Тут ще можна трошки код покращити. Замість того щоб кожного разу рахувати counter оголоси його як поле класу private int counter = 0; І збільшуй його на 1 кожного разу як додав нову пару, тоді тут не буде циклу а в меоді put просто напишеш pairs[counter++] = ....

Reply:
Зроблено. Дякую!